Fix acquiring locks in some admin APIs

donatello opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Some locking bugs were found - these are fixed by:

Acquire:
- read lock on config file during get config
- write lock on config file during update credentials
- write lock on format config

The last item causes HealFormat to be added as an object layer
function.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] All new and existing tests passed.

harshavardhana commented on cmd/admin-handlers.go:
```diff
@@ -575,6 +575,16 @@ func (a adminAPIHandlers) GetConfigHandler(w http.ResponseWriter, r *http.Reques
+		writeErrorResponseJSON(w, ErrOperationTimedOut, r.URL)
+		return
+	}
+	defer configLock.Unlock()
```
This is `configLock.RUnlock()`

donatello replied:
Fixed.

harshavardhana commented on cmd/admin-heal-ops.go:
```diff
@@ -528,19 +528,7 @@ func (h *healSequence) healDiskFormat() error {
-
-	// Heal format.json on available storage.
-	hres, err := healFormatXL(retryingDisks, h.settings.DryRun)
+	hres, err := objectAPI.HealFormat(h.settings.DryRun)
```
No need to add HealFormat() API which will come in in large bucket support.  you only need to hold a lock inside healDiskFormat on `format.json`

donatello replied:
Done.

donatello committed:
Fix locking in some admin APIs:

- read lock for get config
- write lock for update creds
- write lock for format file
